cas-server-support-x509/src/main/java/org/jasig/cas/adaptors/x509/authentication/handler/support/CRLDistributionPointRevocationChecker.java

```diff
@@ -114,47 +114,62 @@ public void setThrowOnFetchFailure(final boolean throwOnFetchFailure) {
      * @see AbstractCRLRevocationChecker#getCRL(X509Certificate)
      */
     @Override
-    protected X509CRL getCRL(final X509Certificate cert) {
+    protected List<X509CRL> getCRLs(final X509Certificate cert) {
         final URI[] urls = getDistributionPoints(cert);
         logger.debug("Distribution points for {}: {}.", CertUtils.toString(cert), Arrays.asList(urls));
+        final List<X509CRL> listOfLocations = new ArrayList<>(urls.length);
+        boolean stopFetching = false;
 
         try {
-            for (final URI url : urls) {
+            for (int index = 0; !stopFetching && index < urls.length; index++) {
+                final URI url = urls[index];
                 final Element item = this.crlCache.get(url);
+
                 if (item != null) {
```

Comment:
Typically, we always check the cache first to see if we have a CRL. If there is one, we simply return that one. If there is not one, we start checking distribution points in the url list and we optionally try them all until we find at least one that has that entry. If we find one, we cache it and return it.